### ISSUE TITLE
Save state between computer restarts/crashes

### ISSUE THREAD

**Issue by seagreen** [xmonad/xmonad#191]
This might be a horrific idea, but I'm typing it from a laptop that crashes occasionally when it sleeps, and it would be awesome if XMonad had a "resume last session" option similar to web browsers.

**Comment by geekosaur** [comment]
Not so much horrific as difficult, unless it's a full desktop environment and all programs "speak" that environment sufficiently. Granting that most programs used with modern desktop environments do speak XDG session protocol — but many of the programs used with xmonad don't even speak the original ICCCM session protocol very well if at all.

**Comment by seagreen** [comment]
Would scoping it to just terminals make it more tractable? XMonad makes it so convenient to spread out that I often end up with 5 or 6 terminals open pretty quickly. Reopening GUI programs manually wouldn't be much of an inconvenience if I could restart terminals automatically.

**Comment by geekosaur** [comment]
Not really, no. How does xmonad or any other program know the difference?

**Comment by seagreen** [comment]
Makes sense. I may take a shot at making a script to do it myself, if I do I'll report how it goes in #xmonad IRC.

In the meantime feel free to close this, since it's not really actionable. Thanks for the help!

**Comment by liskin** [comment]
What about https://linux.die.net/man/1/xsm? I have absolutely no idea if that still works in 2021 but it might.

**Comment by geekosaur** [comment]
Most modern programs don't speak enough of the xsm protocol; they expect an XDG session manager.
